### DISCUSSION TITLE
List user's subscriptions in an organization he is member of

### DISCUSSION THREAD

**Discussion by BiRo70** [community/community#24538] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I’d like to list the organisation subscriptions I am a member of and change their notification settings programmatically.</p>
<pre><code>~$ curl -s -H "Authorization: token ${token}" \
"https://api.github.com/users/${user}/subscriptions?type=all" | jq -c .[] | wc -l
3
</code></pre>
<p>Returns only my subscriptions and not my organisation’s.</p>
<p>I was not able to find anything in the API documentation on how to achieve what I need.</p>
<p>My goal is to list subscriptions and to change the notification setting from “watching” to “releases only”.</p>
<p>Thanks for any help.</p>
<p>r</p>

**Comment by francisfuzz** [comment, marked as answer]
<aside class="quote no-group" data-username="biro70" data-post="1" data-topic="14910">
<div class="title">
<div class="quote-controls"></div>
 biro70:</div>
<blockquote>
<p>I’d like to list the organisation subscriptions I am a member of and change their notification settings programmatically.</p>
<pre><code class="lang-auto">~$ curl -s -H "Authorization: token ${token}" \
"https://api.github.com/users/${user}/subscriptions?type=all" | jq -c .[] | wc -l
3
</code></pre>
</blockquote>
</aside>
<p>:wave: Hey <a class="mention" href="/biro70">@biro70</a>! That’s interesting behavior to report. Would you mind also sharing the response that you get for that request?</p>
<p>When I run that request for your account, I’m able to see a subscription to an organization-owned repositories:</p>
<pre><code class="lang-auto">curl -v -H "Authorization: token $TOKEN" "https://api.github.com/users/biro70/subscriptions?per_page=100"

[
  // ...
  {
    "id": 239826107,
    "node_id": "MDEwOlJlcG9zaXRvcnkyMzk4MjYxMDc=",
    "name": "irules-toolbox",
    "full_name": "f5devcentral/irules-toolbox",
    "private": false,
  // ...
  }
]
</code></pre>
<p>Similarly, I’m able to see subscriptions for my user account for both user-owned and organization-owned repositories:</p>
<pre><code class="lang-auto">curl -v "https://api.github.com/users/francisfuzz/subscriptions?per_page=100"
</code></pre>
<p>I’d like to mention that <a href="https://developer.github.com/v3/activity/watching/#list-repositories-watched-by-a-user">this endpoint</a> doesn’t have any documented input, so <code>?type=all</code> will be ignored here. Also, this only considers subscriptions to public repositories.</p>
<p>If you’re interested in fetching subscriptions for both public and private repositories, it may be worth trying the <a href="https://developer.github.com/v3/activity/watching/#list-repositories-watched-by-the-authenticated-user">endpoint for fetching a list of repositories watched by the authenticated user</a>.</p>

**Comment by BiRo70** [comment]
<p>Hi <a class="mention" href="/francisfuzz">@francisfuzz</a>. Thank you for your reply.<br>
I am quite a novice when it comes to use GitHub API calls.<br>
It was my own fault I couldn’t list the company’s private repositories.<br>
The auth token I generated didn’t have the flag set to access subscriptions.<br>
:man_facepalming:<br>
Anyway all sorted now, also with the help of GitHub support member called Robert.</p>
<p>Thanks!<br>
R</p>
